miner/worker.go

```diff
@@ -391,3 +395,25 @@ func totalFees(block *types.Block, receipts []*types.Receipt) *big.Float {
 	}
 	return new(big.Float).Quo(new(big.Float).SetInt(feesWei), new(big.Float).SetInt(big.NewInt(params.Ether)))
 }
+
+// enforcePredicates takes a set of pending transactions (grouped by sender, and ordered by nonce) and returns a
+// subset of those transactions (grouped by sender) that satisfy predicateContext.
+// Any transaction sent by a given sender that is received after a transaction that does not
+// satisfy predicateContext is removed from the TxPool and is not included in the return value.
+func (w *worker) enforcePredicates(rules params.Rules, predicateContext *contract.PredicateContext, pending map[common.Address]types.Transactions) map[common.Address]types.Transactions {
+	result := make(map[common.Address]types.Transactions, len(pending))
+	for addr, txs := range pending {
+		for i, tx := range txs {
+			if err := core.CheckPredicates(rules, predicateContext, tx); err != nil {
+				log.Debug("Transaction predicate failed verification in miner", "sender", addr, "err", err)
+				w.eth.TxPool().RemoveTx(tx.Hash()) // RemoveTx will move all subsequent transactions back to the future queue
```

Comment:
can we clarify this comment?

Reply:
What would you suggest? Is it just a question of what the future queue within the mempool is that's missing from the comment?

Reply:
Yes, and also what does "subsequent" refer to in this context

Reply:
A subsequent nonce

Reply:
Updated to specify a subsequent nonce, I think if someone needs additional context on the future queue it's reasonable to check `RemoveTx` implementation in the transaction pool rather than elaborate in the comment here.